zerver/tests/test_subs.py

```diff
@@ -359,6 +366,30 @@ def test_make_stream_public(self) -> None:
         self.assertFalse(stream.invite_only)
         self.assertTrue(stream.history_public_to_subscribers)
 
+    def test_make_web_public_stream(self) -> None:
+        user_profile = self.example_user('hamlet')
+        self.login_user(user_profile)
+        realm = user_profile.realm
+        self.make_stream('web_public_stream',
+                         realm=realm,
+                         is_web_public=True,
+                         history_public_to_subscribers=True)
+
+        do_change_is_admin(user_profile, True)
+        params = {
+            'stream_name': ujson.dumps('web_public_stream'),
+            'is_web_public': ujson.dumps(True),
+            'history_public_to_subscribers': ujson.dumps(True)
+        }
+        stream = get_stream('web_public_stream', realm)
+        result = self.client_patch("/json/streams/%d" % (stream.id,), params)
```

Comment:
We should try to use f-strings in the newer code now that we support Python 3.6.